eventuate-common-spring-reactive-jdbc/src/main/java/io/eventuate/common/spring/jdbc/reactive/EventuateSpringReactiveJdbcStatementExecutor.java

```diff
@@ -27,15 +31,22 @@ public EventuateSpringReactiveJdbcStatementExecutor(DatabaseClient databaseClien
   }
 
   public Mono<Long> insertAndReturnId(String sql, String idColumn, Object... params) {
-    sql = String.format("%s%s", reformatParameters(sql, params), ";SELECT LAST_INSERT_ID();");
+    sql = sqlDialect.addReturningOfGeneratedIdToSql(reformatInsertParameters(sql, params, params), idColumn);
```

Comment:
I think we could use the R2DBC method to get the return id.

  public Mono<Long> insertAndReturnId(String sql,
                                      String idColumn,
                                      Object... params) {
    sql = reformatParameters(sql, params);
    DatabaseClient.GenericExecuteSpec genericExecuteSpec = bindParameters(databaseClient.sql(sql)
        .filter(s -> s.returnGeneratedValues(idColumn)), params);

    return genericExecuteSpec
        .map((row, rowMetadata) -> row.get(idColumn, Long.class))
        .first()
        .onErrorResume(t -> Optional.of(t)
            .filter(e -> e instanceof DataIntegrityViolationException)
            .<Mono<Long>>map(e -> Mono.error(new EventuateDuplicateKeyException(e)))
            .orElse(Mono.error(t)));
  }

https://github.com/eventuate-foundation/eventuate-common/issues/97